packages/flame/lib/src/components/position_component.dart

```diff
@@ -1,218 +1,349 @@
-
-  /// X position of this component on the screen (relative to the anchor).
-  double get x => _position.x;
-  set x(double x) => _position.x = x;
-
-  /// Y position of this component on the screen (relative to the anchor).
-  double get y => _position.y;
-  set y(double y) => _position.y = y;
-
-  /// The size that this component is rendered with before [scale] is applied.
-  /// This is not necessarily the source size of the asset.
-  Vector2 get size => _size;
-  set size(Vector2 size) => _size.setFrom(size);
-  final Vector2 _size;
-
-  /// Width (size) that this component is rendered with.
-  double get width => scaledSize.x;
-  set width(double width) => size.x = width / scale.x;
-
-  /// Height (size) that this component is rendered with.
-  double get height => scaledSize.y;
-  set height(double height) => size.y = height / scale.y;
-
-  /// The scale factor of this component
-  Vector2 get scale => _scale;
-  set scale(Vector2 scale) => _scale.setFrom(scale);
-  final Vector2 _scale;
-
-  /// Cache to store the calculated scaled size
-  final ValueCache<Vector2> _scaledSizeCache = ValueCache();
-
-  /// The size that this component is rendered with after [scale] is applied.
-  Vector2 get scaledSize {
-    if (!_scaledSizeCache.isCacheValid([scale, size])) {
-      _scaledSizeCache.updateCache(
-        size.clone()..multiply(scale),
-        [scale.clone(), size.clone()],
-      );
+///   @override
+///   void render(Canvas canvas) {
+///     super.render(canvas);
+///     canvas.drawCircle(Offset(_radius, _radius), _radius, _paint);
+///   }
+/// }
+/// ```
+///
+/// The base [PositionComponent] class can also be used as a container
+/// for several other components. In this case, changing the position,
+/// rotating or scaling the [PositionComponent] will affect the whole
+/// group as if it was a single entity.
+///
+/// The main properties of this class is the [_transform] (which combines
+/// the [position], [angle] of rotation, and [scale]), the [size], and
+/// the [anchor]. Thus, the [PositionComponent] can be imagined as an
+/// abstract picture whose of a certain [size]. Within that picture
+/// a point at location [anchor] is selected, and that point is designated as
+/// a "logical center" of the picture. Then, a sequence of transforms is
+/// applied: the picture is moved so that the anchor becomes at point
+/// [position] on the screen, then the picture is rotated for [angle]
+/// radians around the anchor point, and finally scaled by [scale] also
+/// around the anchor point.
+///
+/// The [size] property of the [PositionComponent] is used primarily for
+/// tap and collision detection. Thus, the [size] should be set equal to
+/// the approximate bounding rectangle of the rendered picture. If you
+/// do not specify the size of a PositionComponent, then it will be
+/// equal to zero and the component won't be able to respond to taps.
+class PositionComponent extends BaseComponent {
+  final Transform2D _transform;
```

Comment:
I have a feeling this Transform2D class ends up being an implementation of the Projector interface I created. it could certainly be used for camera, viewport stuff as well!

Reply:
You're right, `Transform2D` should definitely implement `Projector`. However, if you don't mind, let's defer that to another PR, so that this one wouldn't have to grow even bigger.